### PR TITLE
#55 Implement token-bucket algorithm

### LINKED ISSUE #55: Implement token-bucket algorithm
@lchayoun I'm thinking about to change our main implementation to use a `token-bucket` instead.
Here's a link to take a look https://github.com/bbeck/token-bucket.

WDYT?

### PR DESCRIPTION
**Pull request by lchayoun**
Fixes #55

### CODE REVIEW

**Comment by lchayoun** [comment]
@marcosbarbero please review before I continue to document it :)

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/RateLimiter.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
  */
 public interface RateLimiter {
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/RateLimiter.java]: CRITICAL Move constants to a class or enum. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1214)

**Comment by marcosbarbero** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: spring-cloud-zuul-ratelimit-tests/bucket4j-infinispan/src/main/java/com/marcosbarbero/tests/Bucket4jInfinispanApplication.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+public class Bucket4jInfinispanApplication {
+
+    public static void main(String... args) {
+        SpringApplication.run(Bucket4jInfinispanApplication.class, args);
+    }
+
+    @Bean
+    @Qualifier("RateLimit")
+    public ReadWriteMap<String, GridBucketState> map() {
+        DefaultCacheManager cacheManager = new DefaultCacheManager();
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-tests/bucket4j-infinispan/src/main/java/com/marcosbarbero/tests/Bucket4jInfinispanApplication.java]: BLOCKER Use try-with-resources or close this "DefaultCacheManager" in a "finally" clause. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

**Comment by marcosbarbero** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by lchayoun** [comment]
Saw you did some refactor on the develop branch.
The constant need to move to the constants class on the develop branch.
The other issue is in tests, not critical.

**Comment by marcosbarbero** [comment]
Nice, I didn't notice you were merging into master hehe
From now on let's create all the branches based on develop and then when we have a release we merge to master, WDYT?

**Comment by lchayoun** [comment]
I think that if we work on issue branch and merge to master it will work better.
Combined with what we said about auto release with specific commit message the develop branch is redundant...

**Comment by marcosbarbero** [comment]
Let's follow your approach, I'll delete the develop branch later today and make master the main one again.

**Pull request merged**